modules/standard/IO.chpl

```diff
@@ -2804,111 +3029,139 @@ record defaultSerializer {
+    */
     proc writeValue(const val: ?) throws {
       writer._writeLiteral(": ");
       writer.write(val);
     }
 
-    @chpldoc.nodoc
+    /*
+      Ends serialization of the current map by writing a ``}``
+    */
     proc endMap() throws {
       writer._writeLiteral("}");
     }
   }
-
-  // Map helpers
-  @chpldoc.nodoc
-  proc startMap(writer: fileWriter, size: int) throws {
-    writer._writeLiteral("{");
-    return new MapSerializer(writer);
-  }
-
-  // TODO: How should we handle types that don't have a format-specific
-  // representation, like domains or ranges? Is there a way we can determine
-  // that such types should just be printed inside a string for compatibility?
 }
 
 @deprecated(notes="'DefaultSerializer' is deprecated; please use 'defaultSerializer' instead")
 type DefaultSerializer = defaultSerializer;
 
 /*
-  The default deserializer used by ``fileReader``.
+  The default Deserializer used by ``fileReader``.
 
-  Implements the 'deserializeType' and 'deserializeValue' methods which are
-  called by a ``fileReader`` to deserialize a serialized representation of
-  a type or value.
+  See :ref:`the serializers technote<ioSerializers>` for a general overview
+  of Deserializers and their usage.
 
-  Deserializers may choose to invoke certain user-defined methods or
-  initializers on class and record types. When deserializing a type,
-  deserializers should first attempt to invoke an initializer on the type:
+  Otherwise, please refer to :type:`defaultSerializer` for a description
+  of the default IO format. Individual methods on this type may clarify
+  behavior specific to deserialization
 
-    proc MyType.init(reader: fileReader(?), ref deserializer: reader.deserializerType) throws
+  .. note::
 
-  If this initializer is not available or cannot be resolved, the deserializer
-  may then attempt to invoke a user-defined type method 'deserializeFrom' as a
-  factory method that will produce the type:
+    Prior to the 1.32 release and the advent of the 'serializers' feature, the
+    default implementation for reading classes and records permitted reading
+    fields out of order. This functionality is not present in the
+    ``defaultDeserializer``.
 
-    proc type MyType.deserializeFrom(reader: fileReader(?)
-                                     ref deserializer: reader.deserializerType) : MyType throws
+    For an unspecified amount of time this module will retain the ability to
+    disable automatic use of the ``defaultDeserializer`` by recompiling
+    programs with the config-param ``useIOSerializers`` set to ``false``.
 
-  If neither of these methods are available, the deserializer may attempt to
-  default-initialize the desired type and invoke the value-reading form of
-  deserialization. If the value-reading form of deserialization cannot be
-  resolved, then a compiler error may be issued.
+    Eventually, however, users must update their programs to account for
```

Comment:
I feel like this warning should be moved to the top to account for short attention spans (e.g., mine :D).

Reply:
I don't really see a way to move it to the top since it requires some explaining.

Reply:
Yeah I was thinking more of like a TLDR but I agree it's kind of hard to move.